src/System.Text.Json/src/System/Text/Json/Reader/JsonReaderHelper.cs

```diff
@@ -248,27 +249,69 @@ private static int LocateFirstFoundByte(ulong match)
                                                0x02ul << 40 |
                                                0x01ul << 48) + 1;
 
-        public static bool TryGetEscapedGuid(ReadOnlySpan<byte> span, out Guid value)
+        [MethodImpl(MethodImplOptions.AggressiveInlining)]
+        public static bool IsValidDateTimeParsingLength(int length)
```

Comment:
If DateTime and DateTimeOffset have different "valid lengths", we should split it.

Reply:
They have the same "valid lengths".
Now using "InRange".